lib/rubycritic/commands/compare.rb

```diff
@@ -0,0 +1,203 @@
+# frozen_string_literal: true
+require 'rubycritic/source_control_systems/base'
+require 'rubycritic/analysers_runner'
+require 'rubycritic/revision_comparator'
+require 'rubycritic/reporter'
+require 'rubycritic/commands/base'
+require 'httparty'
+require 'yaml'
+
+module RubyCritic
+  module Command
+    class Compare < Base
+      def initialize(options)
+        super
+        @paths = options[:paths]
+        Config.source_control_system = SourceControlSystem::Base.create
+        @base_branch_hash = {}
+        @feature_branch_hash = {}
+        @files_affected = []
+        @analysed_modules
+        @number
+        @code_index_file_location
+        @app_settings = load_yml
+      end
+
+      def execute
+        compare_branches
+        status_reporter
+      end
+
+      def compare_branches
+        update_build_number
+        set_root_paths
+        switch_to_base_branch_and_compare
+        switch_to_feature_branch_and_compare
+        feature_branch_analysis
+        push_comments_to_svn if Config.merge_request_id
+        compare_code_quality
+      end
+
+      def update_build_number
+        build_file_location = '/tmp/build_count.txt'
+        File.new(build_file_location, "a") unless (File.exist?(build_file_location))
+        @number = File.open(build_file_location).readlines.first.to_i + 1 
+        File.write(build_file_location, @number)
+      end
+
+      def set_root_paths
+        Config.base_root_directory = Pathname.new(base_root_directory)
+        Config.feature_root_directory = Pathname.new(feature_root_directory)
+        Config.build_root_directory = Pathname.new(build_directory)
+      end
+
+      def switch_to_base_branch_and_compare
+        switch_branch(Config.base_branch)
+        critic = critique('base_branch_hash')
+        Config.base_branch_score = critic.score
+        Config.root = base_root_directory
+        Config.base_branch_flag = Config.no_browser = true
+        report(critic)
+        Config.base_branch_flag = false
+      end
+
+      def switch_to_feature_branch_and_compare
+        switch_branch(Config.feature_branch)
+        critic = critique('feature_branch_hash')
+        Config.feature_branch_score = critic.score
+        Config.root = feature_root_directory
+        Config.feature_branch_flag = Config.no_browser = true
+        report(critic)
+        Config.feature_branch_flag = false
+        switch_branch(Config.base_branch)
+      end
```

Comment:
Code duplication in above two methods `switch_to_base_branch_and_compare` and `switch_to_feature_branch_and_compare`